scripts/lib/CIME/XML/compilerblock.py

```diff
@@ -172,19 +173,19 @@ def _handle_references(self, elem, set_up, tear_down, depends):
                 if new_tear_down is not None:
                     tear_down.append(new_tear_down)
                 logger.debug("set_up {} inline {} tear_down {}".format(new_set_up,inline,new_tear_down))
-            elif child.tag == "var":
+            elif self._db.name(child) == "var":
                 # <var> commands also need expansion by the writer, and can
                 # add dependencies.
-                var_name = child.text
+                var_name = self._db.text(child)
                 output += writer.variable_string(var_name)
                 depends.add(var_name)
             else:
                 expect(False,
-                       "Unexpected tag "+child.tag+" encountered in "
+                       "Unexpected tag "+self._db.name(child)+" encountered in "
                        "config_build.xml. Check that the file is valid "
                        "according to the schema.")
-            if child.tail is not None:
-                output += child.tail
+            if child.xml_element.tail is not None:
```

Comment:
Didn't want to think about a replacement for tail, so left in the raw XML call here for now.